pkgs/poetry/poetry-in-venv.nix

```diff
@@ -24,12 +24,29 @@ in
 pkgs.writeShellApplication {
   name = "poetry";
   text = ''
-    # Don't run multiple install workers in parallel if we have only 0.5 CPUs:
-    # helps with speed in free accounts
-    numVCpu=$(${pkgs.jq}/bin/jq .numVCpu </repl/stats/resources.json)
-    if [ "''${numVCpu}" = "0.5" ]; then
+    # Determine how many vcpus we have, first in the standard location, with a fallback
+    # to the cgroup info, since resources.json does not exist during Deployments.
+    if [ -e /repl/stats/resources.json ]; then  # Common resources location
+      numVCpu="$(${pkgs.jq}/bin/jq .numVCpu </repl/stats/resources.json)"
+    else
+      cgroup_path="$(cat /proc/1/cgroup | cut -f 3 -d :)"
```

Comment:
Did you try `cat /proc/1/cgroup` in both Autoscale and Reserved Deployments?

Reply:
Only in Autoscale, I'll give `Reserved` a try, but I was operating under the assumption that we'd always have cgroup limitations, subject to approval by @lincoln-replit

Reply:
I can't think of any reason why it shouldn't work but it would be good to verify that this command works in reserved VMs as well!

Reply:
Pardon the late follow-up. Turns out I am, in fact, falling back to defaults (`0.5` vcpus) due to pid1's cgroup file not existing. I may be misremembering, but I think I heard someone say we just flat out do not run pid1 inside deployment contexts. Is that true? I will keep digging when I come back to this